### PR TITLE
[Azure] Filters service account secret volume mount for Windows

### PR DESCRIPTION
**Pull request by yolo3301**
Fixing issue: https://github.com/virtual-kubelet/virtual-kubelet/issues/40

Also fixing another small bug when kubelet port is not specified in env var. It will use the default port now.

### CODE REVIEW

**Comment by yolo3301** [comment]
@robbiezhang

File: providers/azure/aci.go [hunk 1]
```diff
@@ -713,3 +719,37 @@ func aciContainerStateToContainerState(cs aci.ContainerState) v1.ContainerState
 		},
 	}
 }
+
+// Filters service account secret volumes for Windows.
```
Comment (robbiezhang) [inline providers/azure/aci.go]: please add comments about why we do this.

File: providers/azure/aci.go [hunk 2]
```diff
@@ -713,3 +719,37 @@ func aciContainerStateToContainerState(cs aci.ContainerState) v1.ContainerState
...
+	if strings.EqualFold(osType, "Windows") {
+		serviceAccountSecretVolumeName := ""
+
+		for index, container := range containerGroup.ContainerGroupProperties.Containers {
+			volumeMounts := make([]aci.VolumeMount, 0, len(container.VolumeMounts))
+			for _, volumeMount := range container.VolumeMounts {
+				if !strings.EqualFold(serviceAccountSecretMountPath, volumeMount.MountPath) {
+					volumeMounts = append(volumeMounts, volumeMount)
+				} else {
+					serviceAccountSecretVolumeName = volumeMount.Name
```
Comment (robbiezhang) [inline providers/azure/aci.go]: will it result in multiple volumes?

File: providers/azure/aci.go [hunk 3]
```diff
@@ -713,3 +719,37 @@ func aciContainerStateToContainerState(cs aci.ContainerState) v1.ContainerState
 		},
 	}
 }
+
+// Filters service account secret volumes for Windows.
+func filterServiceAccountSecretVolume(osType string, containerGroup *aci.ContainerGroup) {
+	if strings.EqualFold(osType, "Windows") {
+		serviceAccountSecretVolumeName := ""
```
Comment (robbiezhang) [inline providers/azure/aci.go]: should it be an array?

Comment (yolo3301) [reply providers/azure/aci.go]: Will there be multiple service accounts?

**Comment by robbiezhang** [comment]
LGTM

**Comment by robbiezhang** [comment]
@erikstmartin

**Pull request merged**